src/status_im/extensions/ethereum.cljs

```diff
@@ -72,3 +75,58 @@
                                              (abi-spec/decode (string/replace result-str #"0x" "")  outputs)
                                              :else result-str)]
                             (re-frame/dispatch (on-result (merge {:result result} (when %1 {:error %1})))))]})))
+
+;; eth_getLogs implementation
+
+(defn- event-topic-enc [event params]
+  (let [eventid (str event "(" (string/join "," params) ")")]
+    (abi-spec/sha3 eventid)))
+
+(defn- types-mapping [type]
+  (cond
+    (= "bool" type) :bool
+    (string/starts-with? type "uint") :uint
+    (string/starts-with? type "int") :int
+    (string/starts-with? type "address") :uint
+    (string/starts-with? type "bytes") :bytes
+    (string/starts-with? type "fixed") :bytes
+    :else :bytes))
+
+(defn- values-topic-enc [type values]
+  (let [mapped-type (types-mapping type)]
+    (mapv #(str "0x" (abi-spec/enc {:type mapped-type :value %})) values)))
+
+(defn- parse-topic [t]
+  (cond
+    (nil? t) t ;; nil topic
```

Comment:
Could be simplified with a or?

Reply:
what do you have in mind? I'm still new in clojure :)

Reply:
Something like `(or (nil? t) (string? t) ..) t`